server/broadcast.go

```diff
@@ -507,7 +507,12 @@ func (bsm *BroadcastSessionsManager) chooseResults(ctx context.Context, seg *str
 		// no results from untrusted orch, just using trusted ones
 		return trustedResult.Session, trustedResult.TranscodeResult, trustedResult.Err
 	}
-	segmToCheckIndex := rand.Intn(len(trustedResult.TranscodeResult.Segments))
+	segmcount := len(trustedResult.TranscodeResult.Segments)
+	if segmcount == 0 {
+		err = fmt.Errorf("error transcoding: no transcoded segments in the response from %s", trustedResult.Session.Transcoder())
```

Comment:
just to be safe, could `trustedResult.Session` be nil?

Reply:
Good point. As we can see [here](https://github.com/livepeer/go-livepeer/blob/ccff1f915c11932b6cb1c4c834e31b84fb9fab95/server/broadcast.go#L1089), I think the `trustedResult.Session `couldn't be nil.